demos/https/iot_demo_https_common.c

```diff
@@ -0,0 +1,227 @@
+ * signature in an Authorization header field. We work around this by performing a GET on Range: bytes=0-0. Then
+ * extracting the size of the file from the Content-Range header field in the response. */
+#define RANGE_0_TO_0_HEADER_VALUE            "bytes=0-0"
+#define RANGE_0_TO_0_HEADER_VALUE_LENGTH     ( 9 )
+
+/**
+ * @brief HTTP standard header field "Content-Range"
+ */
+#define CONTENT_RANGE_HEADER_FIELD           "Content-Range"
+#define CONTENT_RANGE_HEADER_FIELD_LENGTH    ( 13 )
+
+/*-----------------------------------------------------------*/
+
+int _IotHttpsDemo_GetS3ObjectFileSize( uint32_t * pFileSize,
+                                       IotHttpsConnectionHandle_t connHandle,
+                                       const char * pPath,
+                                       uint32_t pathLen,
+                                       const char * pAddress,
+                                       uint32_t addressLen,
+                                       uint8_t * pReqUserBuffer,
+                                       uint32_t reqUserBufferLen,
+                                       uint8_t * pRespUserBuffer,
+                                       uint32_t respUserBufferLen )
+{
+    /* Status of HTTPS Client API. */
+    IotHttpsReturnCode_t httpsClientStatus = IOT_HTTPS_OK;
+    /* The HTTPS request configurations for getting the file size. */
+    IotHttpsRequestInfo_t fileSizeReqConfig = { 0 };
+    /* The HTTPS response configurations for getting the file size. */
+    IotHttpsResponseInfo_t fileSizeRespConfig = { 0 };
+    /* Synchronous request specific configurations. */
+    IotHttpsSyncInfo_t reqSyncInfo = { 0 };
+    /* Synchronous response specific configurations. */
+    IotHttpsSyncInfo_t respSyncInfo = { 0 };
+
+    /* Handle identifying the HTTP request. This is valid after the request has been initialized with
+     * IotHttpsClient_InitializeRequest(). */
+    IotHttpsRequestHandle_t fileSizeReqHandle = NULL;
+
+    /* Handle identifying the HTTP response. This is valid after the reponse has been received with
+     * IotHttpsClient_SendSync(). */
+    IotHttpsResponseHandle_t fileSizeRespHandle = NULL;
+
+    /* The status of HTTP response for this request. */
+    uint16_t respStatus = IOT_HTTPS_STATUS_OK;
+
+
+    /* String to store the Content-Range header field value. This header field as we are requesting in this demo is of
+     * the form: "Content-Range: bytes 0-0/FILESIZE", where file size would be the length of the maximum 32 bit integer
+     * which is 10.  Since the header field value "bytes 0-0/FILESIZE" is less than the maximum possible Range header
+     * field value, we size this string to the Range header field value.*/
+    char contentRangeValStr[ RANGE_VALUE_MAX_LENGTH ] = { 0 };
+    /* The location of the file size in the contentRangeValStr. */
+    char * pFileSizeStr = NULL;
+    /* Size in bytes of a single character. */
+    uint8_t sizeOfOneChar = 1;
```

Comment:
Seems like this could instead be sizeof ( uint8_t ) inline below.

Reply:
Thanks